lib/packwerk/cli.rb

```diff
@@ -134,70 +94,40 @@ def update(paths)
     end
 
     def update_deprecations(paths)
-      updating_deprecated_references = ::Packwerk::UpdatingDeprecatedReferences.new(@configuration.root_path)
-      @run_context = Packwerk::RunContext.from_configuration(
-        @configuration,
-        reference_lister: updating_deprecated_references
-      )
-
-      files = fetch_files_to_process(paths)
-
-      @progress_formatter.started(files)
-
-      all_offenses = T.let([], T.untyped)
-      execution_time = Benchmark.realtime do
-        all_offenses = files.flat_map do |path|
-          @run_context.process_file(file: path).tap do |offenses|
-            mark_progress(offenses: offenses, progress_formatter: @progress_formatter)
-          end
-        end
-
-        updating_deprecated_references.dump_deprecated_references_files
-      end
-
-      @out.puts # put a new line after the progress dots
-      show_offenses(all_offenses)
-      @progress_formatter.finished(execution_time)
-      @out.puts("✅ `deprecated_references.yml` has been updated.")
-
-      all_offenses.empty?
+      Commands::UpdateDeprecations.new(
+        out: @out,
```

Comment:
The idea of refactoring the CLI class is to create a thin wrapper around the commands and abstracting some common functionality. [If you have a look at the `DetectStaleViolation` class and you can see that we removed the dependency of `@out` and `@style`](https://github.com/Shopify/packwerk/blob/main/lib/packwerk/cli.rb#L192-L200). Instead, we opted for using a Result object that gets interpreted by the CLI class. This makes the command classes easier to test and reason because it is less complex. 

I would love for this refactor follow the same pattern so that we are consistent 🙂